GVFS/GVFS.Upgrader/UpgradeOrchestrator.cs

```diff
@@ -0,0 +1,375 @@
+                    {
+                        return false;
+                    }
+
+                    this.LogInstalledVersionInfo();
+                    this.LogVersionInfo(newGVFSVersion, newGitVersion, "Available Version");
+
+                    if (!this.preRunChecker.TryRunPreUpgradeChecks(newGitVersion, out errorMessage))
+                    {
+                        return false;
+                    }
+
+                    if (!this.TryDownloadUpgrade(newGVFSVersion, out errorMessage))
+                    {
+                        return false;
+                    }
+
+                    return true;
+                },
+                "Downloading"))
+            {
+                error = errorMessage;
+                return false;
+            }
+
+            if (!this.LaunchInsideSpinner(
+                () =>
+                {
+                    if (!this.preRunChecker.TryUnmountAllGVFSRepos(out errorMessage))
+                    {
+                        return false;
+                    }
+
+                    this.remount = true;
+
+                    if (!this.TryInstallGitUpgrade(newGitVersion, out errorMessage))
+                    {
+                        return false;
+                    }
+
+                    return true;
+                },
+                "Installing Git"))
+            {
+                error = errorMessage;
+                return false;
+            }
+
+            if (!this.LaunchInsideSpinner(
+                () =>
+                {
+                    if (!this.TryInstallGVFSUpgrade(newGVFSVersion, out errorMessage))
+                    {
+                        return false;
+                    }
+
+                    return true;
+                },
+                "Installing GVFS"))
+            {
+                error = errorMessage;
+                return false;
+            }
+
+            this.LogVersionInfo(newGVFSVersion, newGitVersion, "Newly Installed Version");
+
+            newVersion = newGVFSVersion;
+            error = null;
+            return true;
+        }
+        
+        private bool TryRunCleanUp(out string error)
+        {
+            string errorMessage = string.Empty;
+            if (this.remount && !this.LaunchInsideSpinner(
+                () =>
+                {
+                    string remountError;
+                    if (!this.preRunChecker.TryMountAllGVFSRepos(out remountError))
+                    {
+                        EventMetadata metadata = new EventMetadata();
+                        metadata.Add("Upgrade Step", nameof(this.TryRunCleanUp));
+                        metadata.Add("Remount Error", remountError);
+                        this.tracer.RelatedError(metadata, $"{nameof(this.preRunChecker.TryMountAllGVFSRepos)} failed.");
+                        errorMessage += remountError + Environment.NewLine + "You may remount repositories manually with `gvfs mount` from the enlistment.";

```

Comment:
`enlistment` -> `repository`

Reply:
Doesn't the output of `--mount-all` tell us which repos failed to mount? We should show that info to the user, and also include that in the instructions.

But also, if `--mount-all` failed, why will mounting the individual repos succeed? Will this instruction actually be useful for users?

Reply:
>> Doesn't the output of --mount-all tell us which repos failed to mount? We should show that info to the user, and also include that in the instructions.
Yes, we do show that to the user.

>>But also, if --mount-all failed, why will mounting the individual repos succeed? Will this instruction actually be useful for users?
Have removed the additional recovery suggestion message.